plugins/package-managers/bitbake/src/main/kotlin/BitBake.kt

```diff
@@ -0,0 +1,166 @@
+import org.ossreviewtoolkit.plugins.packagemanagers.spdx.SpdxDocumentFile
+import org.ossreviewtoolkit.utils.common.ProcessCapture
+import org.ossreviewtoolkit.utils.common.getCommonParentFile
+import org.ossreviewtoolkit.utils.common.safeDeleteRecursively
+import org.ossreviewtoolkit.utils.common.withoutPrefix
+import org.ossreviewtoolkit.utils.ort.createOrtTempDir
+import org.ossreviewtoolkit.utils.ort.createOrtTempFile
+
+/**
+ * A package manager that uses OpenEmbedded's "bitbake" tool to create SPDX SBOMs [1][2] e.g. for Yocto distributions,
+ * and post-processes these into ORT analyzer results.
+ *
+ * [1]: https://docs.yoctoproject.org/dev/dev-manual/sbom.html
+ * [2]: https://dev.to/angrymane/create-spdx-with-yocto-2od9
+ */
+class BitBake(
+    name: String,
+    analysisRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analysisRoot, analyzerConfig, repoConfig) {
+    class Factory : AbstractPackageManagerFactory<BitBake>("BitBake") {
+        override val globsForDefinitionFiles = listOf("*.bb")
+
+        override fun create(
+            analysisRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = BitBake(type, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    private val scriptFile by lazy { extractResourceToTempFile(BITBAKE_SCRIPT_NAME).apply { setExecutable(true) } }
+    private val spdxConfFile by lazy { extractResourceToTempFile(SPDX_CONF_NAME) }
+
+    private val spdxManager by lazy { SpdxDocumentFile(name, analysisRoot, analyzerConfig, repoConfig) }
+
+    override fun resolveDependencies(definitionFiles: List<File>, labels: Map<String, String>): PackageManagerResult {
+        val commonDefinitionDir = getCommonParentFile(definitionFiles)
+        val workingDir = requireNotNull(commonDefinitionDir.searchUpwardsForFile(INIT_SCRIPT_NAME)) {
+            "No '$INIT_SCRIPT_NAME' script file found for directory '$commonDefinitionDir'."
+        }
+
+        logger.info { "Determined the working directory to be '$workingDir'." }
+
+        val localVersion = getBitBakeVersion(workingDir)
+        val globalVersion = createOrtTempDir().let { dir ->
+            getBitBakeVersion(dir).also { dir.safeDeleteRecursively(force = true) }
+        }
+
+        if (localVersion != globalVersion) {
+            logger.warn { "Local $managerName version $localVersion differs from global version $globalVersion." }
+        }
+
+        val deployDirs = mutableSetOf<File>()
+
+        definitionFiles.forEach { definitionFile ->
+            val target = definitionFile.nameWithoutExtension.substringBeforeLast('_')
+
+            val deployDir = getDeployDir(workingDir, target)
+            deployDirs += deployDir
+
+            val spdxFile = deployDir.findSpdxFiles().find { it.name == "recipe-$target.spdx.json" }
+            if (spdxFile != null) {
```

Comment:
In order to ensure the SPDX file contains the right content, I believe the `if` case should be removed. What do you think ?